### PR TITLE
Utilize bento/core in (p)react extensions

### PR DESCRIPTION
**Pull request by kvchari**
Blocked by https://github.com/ampproject/amphtml/pull/37734
fixes: https://github.com/ampproject/amphtml/issues/37674

### CODE REVIEW

**Review by rileyajones — APPROVED** [review]

**Comment by estherkim** [comment]
is this ready for review?

**Comment by kvchari** [comment]
related concern - @alanorozco I noticed that we're [remap-ing/external-izing all components from src/bento/components](https://github.com/ampproject/amphtml/blob/7550e189a3db9677349415b807fe124cbc25b730/build-system/compile/bento-remap.js#L87) in addition to core modules. I think this will break once we add more bento components to that directory because cross-component dependencies aren't declared in each component's package.json. 

For instance, `bento-stream-gallery` imports `bento-base-carousel`. Currently, this works fine because base-carousel gets bundled with stream-gallery (drawback being that it potentially duplicates it). Since we remap/external-ize cross-component dependencies, stream-gallery will import base-carousel from @bentoproject/base-carousel on npm. But we don't set @bentoproject/base-carousel as a dependency in the package.json (as far as I can tell). Am I correct here?

We could add all the externals that we declare while building a component to that component's package.json's dependencies. But I think that would add every component as a dependency..

**Comment by rileyajones** [comment]
@kvchari I totally agree that we should set peer dependencies in the generated `package.json` files rather than trying to reinvent that wheel. Our current system for generating those files is pretty rudimentary though so that might take a bit of work.

**Comment by kvchari** [comment]
@estherkim yes this is ready for review

File: build-system/tasks/extension-helpers.js [hunk 1]
```diff
@@ -664,14 +652,30 @@ async function buildNpmBinaries(extDir, name, options) {
...
+    for (const mode in npm) {
+      const fullEntryPoint = path.join(extDir, npm[mode].entryPoint);
+      const bentoRemaps = getRemapBentoNpmDependencies(fullEntryPoint);
+      const bentoExternals = Object.values(bentoRemaps);
+      npm[mode].remap = {
+        ...(npm[mode].remap || {}),
+        ...bentoRemaps,
+      };
+      npm[mode].external = [...(npm[mode].external || []), ...bentoExternals];
+    }
```
Comment (estherkim) [inline build-system/tasks/extension-helpers.js:678]: Do you think it's easier to read if `remap` and `external` is set as each mode of npm is declared? 
```
npm = {
      preact: {
        entryPoint: 'component.js',
        outfile: 'component-preact.js',
        external: ['preact', 'preact/dom', 'preact/compat', 'preact/hooks', ...bentoExternals],
        remap: {'preact/dom': 'preact', ...bentoRemaps},
        wrapper: '',
      }
}
```

This way you can avoid some repetition too i.e. calling `getRemapBentoNpmDependencies` for every mode.

Comment (estherkim) [reply build-system/tasks/extension-helpers.js:678]: Oops, I missed that entryPoint is different for each.

Comment (kvchari) [reply build-system/tasks/extension-helpers.js:678]: yeahh had to move it out because they were all different and it was cleaner that way

**Review by estherkim — APPROVED** [review]

**Pull request merged**